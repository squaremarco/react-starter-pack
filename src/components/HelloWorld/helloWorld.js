import React from 'react';
import styled from 'styled-components';

const StyledHeading = styled.h1`
  text-align: center;
  color: #4c4c4c;
`;

export default () => <StyledHeading>Hello, World!</StyledHeading>;
