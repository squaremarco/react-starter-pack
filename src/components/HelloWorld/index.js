import React from 'react';
import Heading from 'Components/StyledComponents/Heading';

const HelloWorld = () => <Heading>Hello, World!</Heading>;

export default HelloWorld;
