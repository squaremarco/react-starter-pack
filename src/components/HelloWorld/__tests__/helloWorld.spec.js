import React from 'react';
import { mount } from 'enzyme';

import HelloWorld from '../helloWorld';

describe('HelloWorld', () => {
  it('Exists', () => {
    const wrapper = mount(<HelloWorld />);
    expect(wrapper.exists()).toEqual(true);
  });

  it('Should render an <h1> element', () => {
    const wrapper = mount(<HelloWorld />);
    expect(wrapper.find('h1').exists()).toEqual(true);
  });

  it('Should render "Hello, World!"', () => {
    const wrapper = mount(<HelloWorld />);
    expect(wrapper.find('h1').text()).toEqual('Hello, World!');
  });

  it('Should match snapshot', () => {
    const wrapper = mount(<HelloWorld />);
    expect(wrapper).toMatchSnapshot();
  });
});
