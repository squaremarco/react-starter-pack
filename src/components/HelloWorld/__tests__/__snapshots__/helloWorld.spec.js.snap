// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Should match snapshot 1`] = `
.c0 {
  text-align: center;
  color: #4c4c4c;
}

<_default>
  <StyledHeading>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "StyledHeading-ea1jpn-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "text-align:center;color:#4c4c4c;",
            ],
          },
          "displayName": "StyledHeading",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "StyledHeading-ea1jpn-0",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h1
        className="c0"
      >
        Hello, World!
      </h1>
    </StyledComponent>
  </StyledHeading>
</_default>
`;
